File: src/com/blackoutburst/bogel/core/MousePositionCallBack.java
package com.blackoutburst.bogel.core;

import org.lwjgl.glfw.GLFWCursorPosCallbackI;

import com.blackoutburst.bogel.maths.Vector2f;

/**
 * <h1>Mouse Position CallBack</h1>
 * 
 * <p>
 * Update mouse position <b>do not call</b>
 * </p>
 * 
 * @since 0.1
 * @author Blackoutburst
 */
public class MousePositionCallBack implements GLFWCursorPosCallbackI {

	@Override
	public void invoke(long window, double xPos, double yPos) {
		Mouse.setX((float) xPos);
		Mouse.setY((float) (Display.getHeight() - yPos));
		Mouse.setPosition(new Vector2f((float) xPos, (float) (Display.getHeight() - yPos)));
	}
}

File: client/src/main/kotlin/dev/blackoutburst/game/utils/MousePositionCallBack.kt
package dev.blackoutburst.game.utils

import dev.blackoutburst.game.core.Display
import dev.blackoutburst.game.maths.Vector2f
import org.lwjgl.glfw.GLFWCursorPosCallbackI

class MousePositionCallBack : GLFWCursorPosCallbackI {
    override fun invoke(window: Long, xPos: Double, yPos: Double) {
        Mouse.x = xPos.toFloat()
        Mouse.y = (Display.getHeight() - yPos).toFloat()
        Mouse.position = Vector2f(xPos.toFloat(), (Display.getHeight() - yPos).toFloat())
    }
}
